mlir/lib/Dialect/SparseTensor/Transforms/SparseTensorRewriting.cpp

```diff
@@ -1148,17 +1148,17 @@ struct ForeachRewriter : public OpRewritePattern<ForeachOp> {
     SmallVector<Value> reducValue = srcBlock->getTerminator()->getOperands();
     rewriter.eraseOp(srcBlock->getTerminator());
 
-    // Inline body.
-    if (!reducValue.empty()) {
-      rewriter.mergeBlocks(srcBlock, rewriter.getBlock(), args);
-    } else {
-      // This is annoying, since scf.for inserts a implicit yield op when
-      // there is no reduction variable upon creation, in this case we need to
-      // merge the block *before* the yield op.
-      rewriter.inlineBlockBefore(srcBlock, &*rewriter.getInsertionPoint(),
-                                 args);
+    Operation &last = rewriter.getBlock()->back();
+    if (llvm::isa<scf::YieldOp>(last)) {
+      // scf.for inserts a implicit yield op when there is no reduction
```

Comment:
(1) Avoid starting comment with what looks like code. For example

Because sc.for inserts .... , we need to merge ....

(2) Change "an implicit yield"

(3)  Make it more clear it applies to code, i.e. we change insertion code to last so that the block is merged before...